x-pack/plugins/secops/public/store/local/network/selectors.ts

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { fromKueryExpression } from '@kbn/es-query';
+import { get } from 'lodash/fp';
+import { createSelector } from 'reselect';
+
+import { State } from '../../reducer';
+import { GenericNetworkModel, NetworkType } from './model';
+
+const selectNetwork = (state: State, networkType: NetworkType): GenericNetworkModel =>
+  get(networkType, state.local.network);
+
+export const topSourceSelector = () =>
+  createSelector(
+    selectNetwork,
+    network => network.queries!.topSource
+  );
+
+export const topDestinationSelector = () =>
+  createSelector(
+    selectNetwork,
+    network => network.queries!.topDestination
+  );
+
+export const networkFilterQueryExpression = () =>
+  createSelector(
+    selectNetwork,
+    network => (network.filterQuery! ? network.filterQuery!.query.expression : null)
+  );
+
+export const networkFilterQueryAsJson = () =>
+  createSelector(
+    selectNetwork,
+    network => (network.filterQuery ? network.filterQuery.serializedQuery : null)
+  );
+
+export const networkFilterQueryDraft = () =>
+  createSelector(
+    selectNetwork,
+    network => network.filterQueryDraft
+  );
+
+export const isNetworkFilterQueryDraftValid = () =>
+  createSelector(
+    selectNetwork,
+    network => {
+      if (network.filterQueryDraft && network.filterQueryDraft.kind === 'kuery') {
+        try {
+          fromKueryExpression(network.filterQueryDraft.expression);
+        } catch (err) {
+          return false;
```

Comment:
Just my opinion on this one (since I'm always on the look out for refactoring repeating patterns and cleaning things up), so no changes required just a suggestion:

I think this pattern is the 3rd time we have the try catch return false or true. Seems like we should just wrap this `fromKueryExpression` in a utility that does the work for us to return a real true or false boolean and we just use that from now on.

These three areas all would clean up to a very nice boolean statement instead. Reading this at the moment is tricky as I'm reading something that returns a value I don't care about but rather I'm just testing to see if it's going to do a throw or not.